Branch-SDK/src/io/branch/referral/BranchPreinstall.java

```diff
@@ -0,0 +1,114 @@
+package io.branch.referral;
+
+import android.content.Context;
+import io.branch.referral.Defines.PreinstallKey;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Iterator;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by --vbajpai on --24/07/2019 at --13:44 for --android-branch-deep-linking-attribution
+ */
+class BranchPreinstall {
+
+    private static final String SYSTEM_PROPERTIES_CLASS_KEY = "android.os.SystemProperties";
+    private static final String BRANCH_PREINSTALL_PROP_KEY = "io.branch.preinstall.apps.path";
+
+    public static void getPreinstallSystemData(Branch branchInstance, Context context) {
+        if (branchInstance != null) {
+            // check if the SystemProperties has the branch file path added
+            String branchFilePath = checkForBranchPreinstallInSystem();
+            if (branchFilePath != null) {
```

Comment:
Because the check returns the empty string, this succeeds.   I think you should call `if (!TextUtils.isEmpty(branchFilePath))`

Reply:
Thanks for the suggestion! :)